### PR TITLE
Clean regression tests

### PR DESCRIPTION
**Pull request by mingwli**
- replace deprovision vsts-docs-pr repo with [windows-compatibility](https://cpubwin.visualstudio.com/_git/windows-compatibility) to make test pass
- add repos list documentation and source excel
- arrange pipeline order to match documentation for easier reference

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6997)

### CODE REVIEW

File: .github/actions/spell-check/dictionary/custom-dictionary.txt [hunk 1]
```diff
@@ -573,3 +573,16 @@ IDid
 IProducer
 msys
 Arity
+csholmq
+dataprep
+devitpro
+generaetd
```
Comment (herohua) [inline .github/actions/spell-check/dictionary/custom-dictionary.txt:579]: `generaetd` is a true type?

**Review by yufeih — COMMENTED** [review]
> arrange pipeline order to match documentation for easier reference
azure pipeline expand matrix in sequential order, the existing order seems to run important tests first.

File: docs/regression-test/regression-test.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+The test baseline is previous **commit build's output**, which is checked in [docfx.testdata](https://ceapex.visualstudio.com/Engineering/_git/docfx.testdata).  
+Each repo has its dedicated branch with the same repo name. e.g. *repo*: [https://github.com/MicrosoftDocs/azure-docs-pr](https://github.com/MicrosoftDocs/azure-docs-pr). *test-data*: [azure-docs-pr](https://ceapex.visualstudio.com/Engineering/_git/docfx.testdata?path=%2F&version=GBazure-docs-pr&_a=contents)
+
+For each PR build, the output will be compared with checked in baseline, and the diff will be commented along with the PR thread if any.
+
+## Selected Docs Repos
+Repos are selected for covering different docfx build functionalities. e.g.: the core markdown parser, yml SDP, versioning & xref, reference plugins, etc.
+
+| page view | repos                         | version | loc | xref | conceptual | SDP conceptual | SDP Ref | multi<br/>docset | Ecma2Yaml | Maml2Yaml | JoinTOC | Split<br/>TOC | Rest | vsts | learn | JS TS | special |
```
Comment (yufeih) [inline docs/regression-test/regression-test.md:17]: How do you propose keeping this table in sync with the matrix inside `azure-pipeline-v3.yml`? The matrix seems like a better place as source of truth.

Comment (mingwli) [reply docs/regression-test/regression-test.md:17]: The matrix lack the purpose of each repo. It's easy to edit the excel and update the markdown accordingly. Besides, this list won't be often updated.

**Comment by mingwli** [comment]
> > arrange pipeline order to match documentation for easier reference
> > azure pipeline expand matrix in sequential order, the existing order seems to run important tests first.

This image shows differently(latter ones already done), I'm not sure about the priority of each task. Not well documented.
image

File: docs/regression-test/regression-test.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Regression Test
+
+## Introduction
+
+Regression test runs docfx e2e build locally against real docs repo to:
+- guard unexptected changes
```
Comment (herohua) [inline docs/regression-test/regression-test.md:6]: `unexptected` is another valid typo

Comment (mingwli) [reply docs/regression-test/regression-test.md]: thanks!

**Review by herohua — APPROVED** [review]
:shipit:

**Review by live1206 — APPROVED** [review]

File: docs/regression-test/regression-test.md (hunk 2 above)
Comment (yufeih) [reply docs/regression-test/regression-test.md:17]: In general it is not a good practice to check in binary files to repos, especially now the same list has replicate 3 places: the matrix/markdown and excel.

Comment (mingwli) [reply docs/regression-test/regression-test.md:17]: Agreed.  We can use *csv* format for easier editing.  

But do you have better solution for achieving following without data duplication?
- have a clear brief view of test repos we have and testing purposes of each repo (simply yml file can't provide this)
- the repo list can be easily updated

**Pull request merged**